File: HNReaderObjC/HNReaderObjC/Story.m
//
//  Story.m
//  HNReaderObjC
//
//  Created by Gorceag Olga on 13/11/16.
//  Copyright © 2016 Gorceag Olga. All rights reserved.
//

#import "Story.h"

NSString *const titleKey = @"title";
NSString *const scoreKey = @"score";
NSString *const timeKey = @"time";
NSString *const authorKey = @"by";
NSString *const commentsCountKey = @"descendants";
NSString *const linkKey = @"url";
NSString *const uniqueIDKey = @"id";

@interface Story ()

@property (nonatomic, strong) NSString *title;
@property (nonatomic, strong) NSString *link;
@property (nonatomic, strong) NSString *author;
@property (nonatomic, strong) NSNumber *timeStamp;
@property (nonatomic, strong) NSNumber *score;
@property (nonatomic, strong) NSNumber *commentsCount;
@property (nonatomic, strong) NSNumber *uniqueID;

@end

@implementation Story

- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
    self = [super init];
    if (self) {
        self.title = dictionary[titleKey];
        self.score = dictionary[scoreKey];
        self.timeStamp = dictionary[timeKey];
        self.author = dictionary[authorKey];
        self.commentsCount = dictionary[commentsCountKey];
        self.link = dictionary[linkKey];
        self.uniqueID = dictionary[uniqueIDKey];
    }
    return self;
}

@end

File: HNReaderSwift/HNReaderSwift/Story.swift
//
//  Story.swift
//  HNReaderSwift
//
//  Created by Gorceag Olga on 16/11/16.
//  Copyright © 2016 Gorceag Olga. All rights reserved.
//

import UIKit

class Story: NSObject {

    let titleKey = "title"
    let scoreKey = "score"
    let timeKey = "time"
    let authorKey = "by"
    let commentsCountKey = "descendants"
    let linkKey = "url"
    let uniqueIDKey = "id"

    var uniqueID: NSNumber!
    var title: String!
    var link: String!
    var author: String!
    var timeStamp: NSNumber!
    var score: NSNumber!
    var commentsCount: NSNumber!

    override init() {
        super.init()
    }

    convenience init(with data:NSDictionary) {
        self.init()
        uniqueID = data[uniqueIDKey] as! NSNumber
        title = data[titleKey] as! String
        score = data[scoreKey] as! NSNumber
        link = data[linkKey] as? String
        author = data[authorKey] as! String
        timeStamp = data[timeKey] as! NSNumber
        commentsCount = data[commentsCountKey] as? NSNumber
    }

}
